atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableCells.java

```diff
@@ -0,0 +1,251 @@
+import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
+import com.palantir.atlasdb.keyvalue.api.ImmutableTargetedSweepMetadata;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RangeRequests;
+import com.palantir.atlasdb.keyvalue.api.RowColumnRangeIterator;
+import com.palantir.atlasdb.keyvalue.api.TargetedSweepMetadata;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.keyvalue.api.WriteReference;
+import com.palantir.atlasdb.schema.generated.SweepableCellsTable;
+import com.palantir.atlasdb.schema.generated.TargetedSweepTableFactory;
+
+public class SweepableCells extends KvsSweepQueueWriter {
+    @VisibleForTesting
+    static final int MAX_CELLS_GENERIC = 50;
+    @VisibleForTesting
+    static final int MAX_CELLS_DEDICATED = 100_000;
+    private static final ColumnRangeSelection ALL_COLUMNS = allPossibleColumns();
+
+    public SweepableCells(KeyValueService kvs, WriteInfoPartitioner partitioner) {
+        super(kvs, TargetedSweepTableFactory.of().getSweepableCellsTable(null).getTableRef(), partitioner);
+    }
+
+    @Override
+    void populateCells(PartitionInfo partitionInfo, List<WriteInfo> writes, Map<Cell, byte[]> cells) {
+        boolean dedicate = writes.size() > MAX_CELLS_GENERIC;
+
+        if (dedicate) {
+            addReferenceToDedicatedRows(partitionInfo, writes, cells);
+        }
+
+        long index = 0;
+        for (WriteInfo write : writes) {
+            addWrite(partitionInfo, write, dedicate, index, cells);
+            index++;
+        }
+    }
+
+    List<WriteInfo> getWritesFromPartition(long partitionFine, ShardAndStrategy shardAndStrategy) {
+        SweepableCellsTable.SweepableCellsRow row = computeRow(partitionFine, shardAndStrategy);
+        RowColumnRangeIterator resultIterator = getColumnRangeAllForRow(row);
+
+        Map<CellReference, WriteInfo> writes = new HashMap<>();
+        resultIterator.forEachRemaining(entry -> populateWrites(row, entry, writes));
+
+        return new ArrayList<>(writes.values());
+    }
+
+    void deleteDedicatedRows(ShardAndStrategy shardAndStrategy, long partitionFine) {
+        rangeRequestsDedicatedRows(shardAndStrategy, partitionFine).forEach(this::deleteRange);
+    }
+
+    void deleteNonDedicatedRow(ShardAndStrategy shardAndStrategy, long partitionFine) {
+        deleteRange(rangeRequestNonDedicatedRow(shardAndStrategy, partitionFine));
+    }
+
+    private void addReferenceToDedicatedRows(PartitionInfo info, List<WriteInfo> writes, Map<Cell, byte[]> cells) {
+        addCell(info, WriteReference.DUMMY, false, 0, -numberDedicatedRows(writes), cells);
+    }
+
+    private void addWrite(PartitionInfo info, WriteInfo write, boolean dedicate, long index, Map<Cell, byte[]> cells) {
+        addCell(info, write.writeRef(), dedicate, index / MAX_CELLS_DEDICATED, index % MAX_CELLS_DEDICATED, cells);
```

Comment:
Clever, but was slightly puzzling why `MAX_CELLS_DEDICATED` was being used for non-dedicated rows. Worth some comments.